### PR TITLE
Add central options and testing framework

### LINKED ISSUE #142: Add testing for tf2autobot
The point of this feature is to add a useful set of tests which can be expanded upon as new features arrive. The test framework is Jest.

Current goals:

  1. Enable asserting the result of SteamUser.chatMessage() calls
  2. Centralize Option loading and default settings
  3. Add mock steam-user implementation

### PR DESCRIPTION
**Pull request by joekiller**
The point of this feature is to add a useful set of tests which can be expanded upon as new features arrive. The test framework is Jest.

Add Centralize Option loading and default settings
Add Testing of option loading
Fixes #142

### CODE REVIEW

**Comment by joekiller** [comment]
Outside of src/lib/DiscordWebhook/userSettings.ts this should be functional.

I'm thinking the options parser could use an optional warning message to print for important missing values

**Comment by joekiller** [comment]
There is one linting warning I need to fix too

**Comment by idinium96** [comment]
`AutoKeys` → `Autokeys`
`CraftWeapon` → `Craftweapon`
`skipBPTFTradeOfferURL` → `skipBPTFTradeofferURL`
`discordWebhookUserName` → `discordWebhookUsername`
`invalidValueExceptionSKUs` → `invalidValueExceptionSKUS`
`customHow2TradeMessage` → `customHow2tradeMessage`

**Comment by joekiller** [comment]
This is ready for review.

**Review by idinium96 — CHANGES_REQUESTED** [review]

File: src/classes/Cart.ts [hunk 1]
```diff
@@ -561,14 +561,14 @@ abstract class Cart {
         delete this.carts[steamID.getSteamID64()];
     }
 
-    static stringify(steamID: SteamID): string {
+    static stringify(steamID: SteamID, disableCraftWeaponAsCurrency: boolean): string {
```
[the remark this answers is not in the document]
Comment (idinium96) [reply src/classes/Cart.ts:564]: oh I see.. this doesn't matter.

File: src/classes/Autokeys/userSettings.ts [hunk 2]
```diff
@@ -1,14 +1,34 @@
...
+    minKeys: number;
+    maxKeys: number;
+    minRefs: number;
+    maxRefs: number;
+}
+
+export function genScrapAdjustment(scrapAdjustmentValue?: number, disableScrapAdjustment?: boolean): ScrapAdjustment {
+    return {
+        enabled: !disableScrapAdjustment,
+        value: !scrapAdjustmentValue ? 0 : scrapAdjustmentValue
```
[the remark this answers is not in the document]
Comment (joekiller) [reply src/classes/Autokeys/userSettings.ts:18]: Is that necessary?
```
> !NaN
true
> !undefined
true
> !""
true
> !null
true
```

[the remark this answers is not in the document]
Comment (idinium96) [reply src/classes/Autokeys/userSettings.ts:18]: ah, I see. alright then. 👌

File: src/classes/Pricelist.ts [hunk 3]
```diff
@@ -557,12 +556,12 @@ export default class Pricelist extends EventEmitter {
...
             const isEnableScrapAdjustmentWithAutoprice =
-                process.env.ENABLE_AUTOKEYS === 'true' &&
-                process.env.DISABLE_SCRAP_ADJUSTMENT === 'false' &&
+                this.options.enableAutokeys &&
+                this.options.disableScrapAdjustment &&
                 currentGlobalKeyBuyingPrice === currentPTFBuyingPrice &&
                 currentGlobalKeySellingPrice === currentPTFSellingPrice;
 
-            if (match === null || (match !== null && match.autoprice) || isEnableScrapAdjustmentWithAutoprice) {
+            if (match === null || match.autoprice || isEnableScrapAdjustmentWithAutoprice) {
```
[the remark this answers is not in the document]
Comment (joekiller) [reply src/classes/Pricelist.ts:564]: seems safe in nodejs 12 but I'll include it

```
const match = null;
> if (match === null || match.autoprice || bob.uncle) { console.log("nooo"); }
nooo
undefined
>
```

**Pull request merged**